LPS-131512 Creating a User Group in a Publication causes pre-existing User Groups from Production to be filtered out

ctang3 opened the pull request:
https://issues.liferay.com/browse/LPS-131512

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

Preston-Crary commented:
Please also include a integration test in SearchCTTest.

ctang3 commented:
@Preston-Crary finished with adding an integration test for the bug fix. 

In https://github.com/Preston-Crary/liferay-portal/pull/1332/commits/cf6f13e7ad80f6cc9116b442d91867bf5cce48e2 I split asserting production hits into its own method so it reads more clearly that its asserting production instead of a CTCollection and also so that I could modify `assertCollectionHits` to assert CTModels from production in addition to those from a CTCollection. 

For https://github.com/Preston-Crary/liferay-portal/pull/1332/commits/3e050d29f5929b99aab22193985db1d69f00ae89 I might be wrong, but it appeared that `testPublishAndUndoArticle` and `testPublishAndUndoLayout` weren't actually undoing/reverting the CTCollections created in those tests and the final assertions weren't asserting the models from the original production state from before the CTCollection was published (they asserted the same models as the as post-publish pre-undo assertion). `_undoCTCollection` was being created but without  `_ctProcessLocalService.addCTProcess` being called on it after so I added that.

ctang3 commented on modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/search/test/SearchCTTest.java:
```diff
@@ -201,28 +203,35 @@ public void testPublishAndUndoArticle() throws Exception {
 			deletedJournalArticle = _journalArticleLocalService.deleteArticle(
 				deletedJournalArticle);
 
-			modifiedJournalArticle2 = JournalTestUtil.updateArticle(
-				modifiedJournalArticle1, "testModifyJournalArticle");
```
I noticed that if a `JournalArticle` is updated using `JournalTestUtil.updateArticle` , undoing/reverting the published CTCollection in this testcase removed `modifiedJournalArticle` from the production index instead of it being reverted  back to its pre-published state. Using `_journalArticleLocalService` following the update pattern in  `testPublishAndUndoLayout` results in the correct expected articles being present after the CTCollection is reverted.

Preston-Crary replied:
Yes, this is the bug I was talking about earlier.  Technically using JournalArticleLocalService.update(JournalArticle) is wrong because that's not the method used by the user when updating an article.
